<launch>
  <param name="/use_sim_time" value="true"/>
  
  <!-- Map server -->
  <node name="map_server" pkg="map_server" respawn="false" type="map_server" args="$(find experiment_launcher)/map/map.yaml">
    <param name="frame_id" value="map"/>
  </node>

  <group ns="robot1">
    <param name="tf_prefix" value="robot1_tf" />
      
    <node name="move_base_node" output="screen" pkg="move_base" respawn="true" type="move_base">
      <param name="oscillation_timeout" value="10"/>
      <param name="controller_patience" value="5"/>
      <param name="clearing_rotation_allowed" value="false"/>
      <!--      <param name="recovery_behaviors" value="[{name: conservative_reset, type: clear_costmap_recovery/ClearCostmapRecovery}, {name: aggressive_reset, type: clear_costmap_recovery/ClearCostmapRecovery}]" />-->
      <rosparam command="load" file="$(find experiment_launcher)/config/costmap_common_params.yaml" ns="global_costmap"/>
      <rosparam command="load" file="$(find experiment_launcher)/config/costmap_common_params.yaml" ns="local_costmap"/>
      <rosparam command="load" file="$(find experiment_launcher)/config/local_costmap_params.yaml"/>
      <rosparam command="load" file="$(find experiment_launcher)/config/global_costmap_params.yaml"/>
      <rosparam command="load" file="$(find experiment_launcher)/config/base_local_planner_params.yaml"/>

      <remap from="map" to="/map"/>
    </node>

    <param name="initial_pose_x" value="1"/> 
    <param name="initial_pose_y" value="1"/> 
    <param name="initial_pose_a" value="0"/>

    <param name="amcl/initial_pose_x" value="1"/>
    <param name="amcl/initial_pose_y" value="1"/>
    <param name="amcl/initial_pose_a" value="0"/>
    <param name="amcl/odom_frame_id" value="/robot1/odom"/>
    <param name="amcl/base_frame_id" value="/robot1/base_link"/>
    <param name="amcl/global_frame_id" value="/map"/>

    <include file="$(find experiment_launcher)/launch/nested_amcl/amcl_node.xml"/>
  </group>

  <!-- <group ns="robot2">
    <param name="tf_prefix" value="robot2_tf" />
    <param name="initial_pose_x" value="-1" />
    <param name="initial_pose_y" value="1" />
    <arg name="initial_pose_a" value="0.0"/>
    <include file="$(find experiment_launcher)/launch/nested_amcl/run_nested_amcl.launch" />
  </group> -->

  <!-- <node pkg="rviz" type="rviz" name="rviz" args="-d $(find experiment_launcher)/config/multi.vcg"   output="screen" /> -->

</launch>